ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -2973,17 +3012,28 @@ static void ggml_sycl_mul_mat(ggml_backend_sycl_context & ctx, const ggml_tensor
         // KQ + KQV multi-batch
         ggml_sycl_mul_mat_batched_sycl(ctx, src0, src1, dst);
     } else if (use_dequantize_mul_mat_vec) {
-        opt_for_reorder(&ctx, src0, src1, dst); //the OP function in this branch support reorder.
-        ggml_sycl_op_mul_mat(ctx, src0, src1, dst, ggml_sycl_op_dequantize_mul_mat_vec, false);
-        // save_tensor_txt("1/dst_1.txt", (float*) dst->data, src0->ne[1], sizeof(float), ctx.stream());
+        constexpr bool convert_src1_to_q8_1 = false;
+        if (ggml_sycl_supports_reorder_dmmv(src0->type)) {
+            opt_for_reorder(&ctx, src0, src1, dst);
+        }
```

Comment:
I suggest only call opt_for_order() here, instead of call 2 functions.

Suggest like:

opt_for_reorder(&ctx, src0, src1, dst, mm_type = MM_TYPE_DMMV) {
  if (mm_type == MM_TYPE_DMMV) {
        if (src0->type!=Q4_0) return;     
  }
  if (mm_type == MM_TYPE_MMVQ) {
        if (src0->type!=Q4_0) return;     
  }
  ....
}

Reply:
Same comments for following similar code.

Reply:
It's great!